src/main/java/com/dataguise/saas/service/ConnectionService.java

```diff
@@ -0,0 +1,450 @@
+import com.dg.saas.orch.exception.DgDrvException;
+import com.dg.saas.orch.models.structures.Connection;
+import com.dg.saas.orch.models.structures.ErrorConstants;
+import io.swagger.client.model.*;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * The Connection service
+ */
+@Service
+public class ConnectionService {
+
+    private static Logger logger = Logger.getLogger(ConnectionService.class);
+
+    @Value("${dgsecure.scheduleScanType}")
+    private String scanType;
+
+    @Value("${dgsecure.scheduleStartDelayInMins}")
+    private Integer delayInMins;
+
+    @Autowired
+    private DgSecureRestDriver restDriver;
+
+    @Autowired
+    private SchedulerService schedulerService;
+
+    @Autowired
+    private TaskBuilderService taskBuilderService;
+
+    @Autowired
+    private AgentService agentService;
+
+
+    /**
+     * Gets connections for both detection and protection in a flat structure.
+     *
+     * @return List of all connections in flat structure.
+     */
+    public List<ConnectionDTO> getConnections(Optional<Integer> id) throws BadGatewayException {
+        List<DgConnection> dgConnections;
+        List<ConnectionDTO> connectionsUIFormatter = new ArrayList<>();
+        try {
+            dgConnections = restDriver.getAllConnections();
+        } catch (DgDrvException e) {
+            throw new BadGatewayException(e.getMessage(), HttpStatus.BAD_GATEWAY);
+        }
+        if (id.isPresent()) {
+            dgConnections = dgConnections.stream().
+                filter(dgConnection -> dgConnection.getConnectionId().equals(id.get()))
+                .collect(Collectors.toList());
+        }
+        if (dgConnections != null && !dgConnections.isEmpty()) {
+            for (DgConnection dgConnection : dgConnections) {
+                ConnectionDTO connection = new ConnectionDTO();
+                Integer groupId = dgConnection.getGroupId();
+                List<DgInfraStructureNodeWrapper> agentDetails = agentService
+                    .filterAgentListByGroupId(groupId);
+                DgInfraStructureNodeWrapper agent = null;
+                if (agentDetails != null && !agentDetails.isEmpty()) {
+                    agent = agentDetails.get(0);
+                    connection.setAgentHostName(agent.getHostName());
+                    connection.setAgentName(agent.getAgentName());
+                    connection.setConnection(dgConnection);
+                    connection.setAutoSchedule(null);
+                    connectionsUIFormatter.add(connection);
+                } else {
+                    logger.warn("Unable to fetch agent details for connectionId: " + dgConnection
```

Comment:
**Reporter**: PMD    
    **Rule**: GuardLogStatement
    **Severity**: ERROR
    **File**: /work/src/main/java/com/dataguise/saas/service/ConnectionService.java L83
    
    Logger calls should be surrounded by log level guards.&#10;&#10;&#10;Best Practices https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement
    
*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1912301417>*